rotkehlchen/types.py

```diff
@@ -956,15 +980,43 @@ def serialize(self) -> dict[str, str | None]:
             'blockchain': self.blockchain.serialize() if self.blockchain is not None else None,
         }
 
+    @staticmethod
+    def check_chain_ecosystem(address: 'BlockchainAddress') -> Literal[
+        ChainType.BITCOIN,
+        ChainType.EVMLIKE,
+        ChainType.SUBSTRATE,
+    ]:
+        """
+        Get the ecosystem for the provided address being it either EVMLIKE, Bitcoin or Substrate.
+        TODO: Add solana in develop
+
+        May raise: InputError
+        """
+        if is_checksum_address(address):
+            return ChainType.EVMLIKE
+        elif is_valid_btc_address(address) or is_valid_bitcoin_cash_address(address):
+            return ChainType.BITCOIN
+        elif (
+            is_valid_ss58_address(value=address, valid_ss58_format=0) or  # Polkadot
+            is_valid_ss58_address(value=address, valid_ss58_format=2)     # Kusama
+        ):
+            return ChainType.SUBSTRATE
+
+        raise InputError('Unknown address chainType when identifying ecosystem')
+
+    @staticmethod
+    def get_ecosystem_key_by_address(address: 'BlockchainAddress') -> str:
+        ecosystem = AddressbookEntry.check_chain_ecosystem(address)
+        return f'{ADDRESSBOOK_BLOCKCHAIN_GROUP_PREFIX}{ecosystem.name}'
+
+    def get_ecosystem_key(self) -> str:
```

Comment:
no need for this. we can just use the static method directly right?

Reply:
this is here just to avoid needing to pass the self.address as argument